### PR TITLE
Use PGFuncGeneratorInfo to codegen pg functions

### PR DESCRIPTION
**Pull request by karthijrk**
I am using PGFuncGeneratorInfo struct to pass around information needed for pg_func code generation. 

@foyzur @hardikar please have a look when you get chance.

### CODE REVIEW

File: src/backend/codegen/include/codegen/pg_func_generator.h [hunk 1]
```diff
@@ -182,20 +180,21 @@ Arg0, Arg1> {
...
     std::vector<llvm::Value*> llvm_preproc_args;
-    if (!this->PreProcessArgs(codegen_utils, llvm_args, &llvm_preproc_args)) {
+    if (!this->PreProcessArgs(codegen_utils,
+                              pg_func_info.llvm_args,
+                              &llvm_preproc_args)) {
       return false;
     }
+    PGFuncGeneratorInfo pg_processed_func_info(pg_func_info.llvm_main_func,
+                                               pg_func_info.llvm_error_block,
+                                               std::move(llvm_preproc_args));
```
Comment (hardikar) [inline src/backend/codegen/include/codegen/pg_func_generator.h]: This invalidates llvm_preproc_args after this point.

File: src/backend/codegen/op_expr_tree_generator.cc [hunk 2]
```diff
@@ -186,11 +186,12 @@ bool OpExprTreeGenerator::GenerateCode(GpCodegenUtils* codegen_utils,
...
   llvm::Value* llvm_op_value = nullptr;
-  bool retval = pg_func_interface->GenerateCode(codegen_utils,
-                                   gen_info.llvm_main_func,
+  PGFuncGeneratorInfo pg_func_info(gen_info.llvm_main_func,
                                    gen_info.llvm_error_block,
-                                   llvm_arguments,
-                                   &llvm_op_value);
+                                   std::move(llvm_arguments));
+  bool retval = pg_func_interface->GenerateCode(codegen_utils,
+                                          pg_func_info,
```
Comment (hardikar) [inline src/backend/codegen/op_expr_tree_generator.cc]: indentation

**Comment by hardikar** [comment]
Minor comments. LGTM otherwise.

File: src/backend/codegen/include/codegen/pg_func_generator.h (hunk 1 above)
Comment (karthijrk) [inline src/backend/codegen/include/codegen/pg_func_generator.h]: Will remove move and made constructor to take const ref.

**Comment by karthijrk** [comment]
Addressed PR comments

**Pull request merged**